File: backend/entity/Company.go
package entity

import (
	"gorm.io/gorm"
)

type Company struct {
	gorm.Model
	CompanyName string `json:"company_name"`
	Logo        string `json:"logo"`

	UserID uint `json:"user_id"`
	User   User `gorm:"foreignKey:UserID"`

	AddressID uint    `json:"address_id"`
	Address   Address `gorm:"foreignKey:AddressID"`

	ContactID uint    `json:"contact_id"`
	Contact   Contact `gorm:"foreignKey:ContactID"`

	IntershipPosts        []IntershipPost        `gorm:"foreignKey:CompanyID"` // one-to-many
	InterviewAppointments []InterviewAppointment `gorm:"foreignKey:CompanyID"` // one-to-many
	Reviews               []Review               `gorm:"foreignKey:CompanyID"`
}

File: frontend/src/interfaces/Company.ts
import type { AddressInterface } from "./Address";
import type { AdminInterface } from "./Admin";
import type { UserInterface } from "./User";
import type { ContactInterface } from "./Contact";
import type { IntershipPostInterface } from "./IntershipPost";
import type { InterviewAppointmentInterface } from "./InterviewAppointment";
import type { ReviewInterface } from "./Review";
import type { VerifyInterface } from "./Verify";

export interface CompanyInterface {
  data: CompanyInterface | PromiseLike<CompanyInterface>;
  ID?: number;
  company_name: string;
  logo: string;

  user_id: number;
  User?: UserInterface;

  address_id: number;
  Address?: AddressInterface;

  admin_id: number;
  Admin?: AdminInterface;

  Contact?: ContactInterface;
  Verify?: VerifyInterface;
  Intership_posts?: IntershipPostInterface[];
  Interview_appointments?: InterviewAppointmentInterface[];
  Reviews?: ReviewInterface[];
}
